core/slice.go

```diff
@@ -235,6 +243,12 @@ func (sl *Slice) ConstructLocalBlock(header *types.Header) *types.Block {
 			for i, blockHash := range pendingBlockBody.SubManifest {
 				subBlockHashes[i] = blockHash
 			}
+			if subBlockHashes.Hash() != header.ManifestHash() {
+				subManifest := sl.getPendingSubManifest(header.ManifestHash())
+				if subManifest != nil {
+					subBlockHashes = subManifest
```

Comment:
what happens if it is nil even after this ? Is there a point in continuing.

Reply:
I now return `nil` to indicate an invalid block. Does that seem sufficient?